src/loki/search.cc

```diff
@@ -498,14 +498,11 @@ struct bin_handler_t {
 
       // if this edge is filtered
       const auto* edge = tile->directededge(edge_id);
-      if (edge_filter(edge) == 0.0f) {
+      if (costing->Filter(edge, edge_id, tile) == 0.0f) {
         // then we try its opposing edge
-        edge_id = reader.GetOpposingEdgeId(edge_id, tile);
+        edge_id = reader.GetOpposingEdgeId(edge_id, edge, tile);
         // but if we couldnt get it or its filtered too then we move on
-        if (!edge_id.Is_Valid())
-          continue;
-        edge = tile->directededge(edge_id);
-        if (edge_filter(edge) == 0.0f)
+        if (!edge_id || costing->Filter(edge, edge_id, tile) == 0.0f)
```

Comment:
i think this is a bug. doing `!edge_id` will implicit cast edge_id to uint64_t. the boolean cast needs to be explicit for it to automatically call `Is_Valid` internally. i think just in case you should call `Is_Valid` maybe in the future we should just delete the explicit boolean cast operator in graphid just to avoid having to think about this